### PR TITLE
lightning: refactor checksum to reuse

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #42930 

Problem Summary:

### What is changed and how it works?
- move checksum code into separate pkg

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/importer/checksum_helper.go
Comment (D3Hunter) [inline br/pkg/lightning/importer/checksum_helper.go:1]: this file named as `checksum_helper` to make git recongnize `local/checksum.go` as rename, so less code change

File: br/pkg/lightning/importer/tidb.go [hunk 1]
```diff
@@ -234,30 +234,6 @@ func LoadSchemaInfo(
 	return result, nil
 }
 
-// ObtainGCLifeTime obtains the current GC lifetime.
-func ObtainGCLifeTime(ctx context.Context, db *sql.DB) (string, error) {
```
Comment (D3Hunter) [inline br/pkg/lightning/importer/tidb.go:238]: moved to checksum.go

**Comment by D3Hunter** [comment]
ci pass https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_integration_br_test/detail/tidb_ghpr_integration_br_test/12532/pipeline

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by gozssky — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**